src/components/UploadAvatarModal/UploadAvatarModal.js

```diff
@@ -0,0 +1,247 @@
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import PropTypes from 'prop-types'
+import Slider from 'rc-slider'
+import { useState, useCallback } from 'react'
+import Cropper from 'react-easy-crop'
+import { useDispatch } from 'react-redux'
+
+import asModal, { ModalContent, ModalFooter } from '~/components/asModal'
+import { updateAvatar } from '~/store/actions/user'
+import getResponseError from '~/util/getResponseError'
+
+import UploadAvatarMessageBox from './UploadAvatarMessageBox'
+import styles from './UploadAvatarModal.module.scss'
+
+// Component Constants
+const MAX_FILE_SIZE = 26214400 // Server upload max is 25M
+const SUBMIT_AUTO_CLOSE_DELAY_TIME = 3000
+const HALF_CIRCLE = 180 // Conversion of rat to deg
+
+
+function UploadAvatarModal (props) {
+  const {
+    onClose,
+    isOpen,
+  } = props
+
+  const [result, setResult] = useState({})
+
+  const [upImg, setUpImg] = useState()
+  const [crop, setCrop] = useState({ x: 0, y: 0 }) // Lint comment on "Identifier name is too short" - this is required by the Cropper. Cannot change.
+  const [zoom, setZoom] = useState(1)
+  const [rotation, setRotation] = useState(0)
+  const [croppedAreaPixels, setCroppedAreaPixels] = useState(null)
+  const [submitReady, setSubmitReady] = useState(false)
+  const [submitting, setSubmitting] = useState(false)
+
+  // Linting note: Cannot use arrow function in onevent, need to wrap
+  const handleSetCrop = useCallback((newCrop) => {
+    setCrop(newCrop)
+  }, [])
+  const handleSetZoom = useCallback((newZoom) => {
+    setZoom(newZoom)
+  }, [])
+  const handleSetRotation = useCallback((newRotation) => {
+    setRotation(newRotation)
+  }, [])
```

Comment:
This is a useless wrapping of `setState` funcs. All of these callbacks could be bound directly to the event listener props. for instance:

```jsx
const [crop, handleCropChange] = useState({ x: 0, y: 0 })

/* ... */

return (<Cropper onCropChange={handleCropChange} />)
```

Reply:
Umm... I tried that to start with, but lint complained bitterly that I wasn't allowed... I 100% agree it's useless otherwise...

Reply:
This is very frustrating - I'd not wrapped the functions previously but something was broken with the way you have written it before - either lint complained or the sliders simply stopped working, but changing them back just now they're working. I have no idea why. Because I 100% agree with you that such a wrap was useless.

Oh well.